packages/iov-ethereum/src/ethereumconnection.spec.ts

```diff
@@ -1017,19 +1113,244 @@ describe("EthereumConnection", () => {
       })().catch(done.fail);
     }, 60_000);
 
-    it("can listen to transactions by ID (transaction in history)", done => {
+    it("works for ERC20 transactions by recipient (in history and updates)", done => {
+      pendingWithoutEthereum();
+
+      (async () => {
+        const codec = new EthereumCodec({ erc20Tokens: testConfig.erc20Tokens });
+        const connection = await EthereumConnection.establish(testConfig.base, {
+          ...testConfig.connectionOptions,
+          erc20Tokens: testConfig.erc20Tokens,
+        });
+
+        const recipientAddress = await randomAddress();
+
+        // send transactions
+
+        const profile = new UserProfile();
+        const wallet = profile.addWallet(Secp256k1HdWallet.fromMnemonic(testConfig.mnemonic));
+        const sender = await profile.createIdentity(wallet.id, testConfig.chainId, HdPaths.ethereum(0));
+
+        const sendA: SendTransaction = {
+          kind: "bcp/send",
+          creator: sender,
+          recipient: recipientAddress,
+          fee: {
+            gasPrice: testConfig.gasPrice,
+            gasLimit: testConfig.gasLimit,
+          },
+          ...testConfig.erc20TransferTests[0],
+        };
+
+        const sendB: SendTransaction = {
+          kind: "bcp/send",
+          creator: sender,
+          recipient: recipientAddress,
+          fee: {
+            gasPrice: testConfig.gasPrice,
+            gasLimit: testConfig.gasLimit,
+          },
+          ...testConfig.erc20TransferTests[0],
+        };
+
+        const sendC: SendTransaction = {
+          kind: "bcp/send",
+          creator: sender,
+          recipient: recipientAddress,
+          fee: {
+            gasPrice: testConfig.gasPrice,
+            gasLimit: testConfig.gasLimit,
+          },
+          ...testConfig.erc20TransferTests[0],
+        };
```

Comment:
Does this really need to be defined three times?

Reply:
All 3 are indeed equal in this test. Given that there is no nice native object cloning, I prefer to roll out this code.

Reply:
I was assuming they were going to be consumed by a pure function, but maybe that's not a safe assumption to make.